src/server/ua_server_binary.c

```diff
@@ -23,6 +23,13 @@
 #include "ua_server_internal.h"
 #include "ua_types_encoding_binary.h"
 #include "ua_services.h"
+#include "mp_printf.h"
+
+#ifdef _WIN32
+# include <winsock2.h>
```

Comment:
We don't include system libraries from the /src folder. Only the plugins do it.
Move the DiscoveryUrl generation and `UA_Array_appendCopy` into the `serverNetworkCallback`.
There the eventloop will give you the hostnames for listen sockets when they open.